crates/rooch/src/commands/move_cli/commands/run_function.rs

```diff
@@ -81,15 +83,24 @@ impl CommandAction<ExecuteTransactionResponseView> for RunFunction {
             })
             .collect::<Result<Vec<_>>>()?;
         let action = MoveAction::new_function_call(function_id, type_args, args);
-        match (self.tx_options.authenticator, self.tx_options.session_key) {
+
+        let dry_run_result = context
```

Comment:
If the `dry_run_result` is an error, we should return the result and not try to execute the tx.